### ISSUE TITLE
Where can I update the configuration file?

### ISSUE THREAD

**Issue by ahmedhamdy90** [provectus/kafka-ui#706]
Hello there,  I am sorry for asking a naive question, but where can I update the configuration file (application-local.yml) for the docker to add my clusters?  Thank you so much :)

**Comment by Haarolean** [comment]
Hi,

you could override/add such settings via environment variables. We have a bunch of example docker-compose files, for example please take a look at `kafka-ui.yaml` file.

**Comment by germanosin** [comment]
Hi, @ahmedhamdy90. Thx for asking questions) 
Simplest way is to pass parameters by env variables, like KAFKA_CLUSTERS_0_NAME, you can find them in our Readme. We are using spring config, so any parameter could be overridden by env variables.

**Comment by ahmedhamdy90** [comment]
I don't know where can I find that file? And how can I locate `application-local.yml` file as it is mentioned in the `readme` file? Like is it something I can access from Docker GUI or Terminal?  I am on MacOS.

**Comment by Haarolean** [comment]
First of all, how do you run the app?

**Comment by ahmedhamdy90** [comment]
I run the app as from your docker image, then using the command in the read me file, and it is working fine. However, in the read me, it is mentioned that I can use the application-local.yml for easier startup configuration.

**Comment by Haarolean** [comment]
Ah, got it. I guess you run it via `docker run` then.
Then I suggest you run it via `docker-compose` since it's way easier.
There's a bunch of example files in repo, e. g. `kafka-ui.yaml`. You just
have to edit the file and run `docker-compose` from the same directory your
compose file is located.
Please refer to this link as a reference: https://docs.docker.com/compose/

пт, 23 июл. 2021 г., 18:55 Ahmed Hamdy ***@***.***>:

**Comment by rmrf** [comment]
@Haarolean  I have same question, does the Env is the only way to setup configuration for kafka-ui ? can I use configuration file?

**Comment by Haarolean** [comment]
@rmrf Yes you can: https://www.reddit.com/r/AskReddit/comments/1g9kel/whats_the_weirdest_thing_your_so_asked_you_to_do/

**Comment by rmrf** [comment]
> @rmrf Yes you can: https://github.com/provectus/kafka-ui#configuration-file

I can not figure out where to place this file, then kafka-ui can use this file, same as @ahmedhamdy2121  asking

**Comment by Haarolean** [comment]
> > @rmrf Yes you can: https://github.com/provectus/kafka-ui#configuration-file
> 
> I can not figure out where to place this file, then kafka-ui can use this file, same as @ahmedhamdy2121 asking

wherever you want actually. You just have to pass the arg like this: `java -Dspring.config.additional-location=/tmp/conf.yml -jar kafka-ui-api...jar`

**Comment by rmrf** [comment]
@Haarolean Thank you! it will be great if document have this.

**Comment by Haarolean** [comment]
> @Haarolean Thank you! it will be great if document have this.

@rmrf you can raise a PR, we have too little bandwidth for documentation rn :(
